### PR TITLE
feat: add Matija's Homepage realm to examples

### PR DESCRIPTION
**Pull request by matijamarjanovic**
### Summary
This pull request adds a new realm example to the Gno `examples` repository—Matija's Homepage. It showcases a personal homepage built on the Gno chain where users can interact by voting with GNOT tokens to change the page's color. The more tokens users send, the greater influence they have on the color scheme, providing an interactive and dynamic experience.

### Key Features
- **Profile Section**: Displays a personal profile with an image and description.
- **Color Voting**: Users can vote for the page's color (red, green, blue) by sending GNOT tokens. RGB values are adjusted based on the amount sent.
- **Dynamic Updates**: The homepage dynamically updates the color based on votes, showcasing real-time interaction on the Gno blockchain.
- **Links to GitHub and LinkedIn**: Includes buttons for GitHub and LinkedIn, making it easy for users to connect.

### Tools & Technologies
- Utilizes Gno's native functions to handle voting and token transfers.
- Provides a simple, yet effective example of how personal realms can be interactive and engaging on the Gno platform.

### Why this is valuable
This example highlights the possibilities of personal realms on Gno, showing how users can create unique and interactive profiles. It’s a fun and approachable entry point for anyone new to Gno development, while also demonstrating the platform's flexibility and potential for creative expression.

### CODE REVIEW

File: examples/gno.land/r/matijamarjanovic/config/config.gno
Comment (notJoon) [inline examples/gno.land/r/matijamarjanovic/config/config.gno:10]: these two variable's names are seems ambiguous. Could you update their names? like `mainAddr`

File: examples/gno.land/r/matijamarjanovic/config/config.gno
Comment (notJoon) [inline examples/gno.land/r/matijamarjanovic/config/config.gno:27]: error messages are repeated (invalid arress and unauthorized). please make them as constant.

File: examples/gno.land/r/matijamarjanovic/config/config.gno
Comment (notJoon) [inline examples/gno.land/r/matijamarjanovic/config/config.gno:25]: very niptick: consider change parameter name `a` to more concrete name.

**Review by notJoon — CHANGES_REQUESTED** [review]
Left some comments. please check them out. thank you

**Comment by matijamarjanovic** [comment]
Thank you for your insights. Fixed the issues, please let me know if I can make anything more appropriate.

File: examples/gno.land/r/matijamarjanovic/config/config.gno
Comment (matijamarjanovic) [reply examples/gno.land/r/matijamarjanovic/config/config.gno:10]: Hi, please see fix in this commit: 16529d3bbe583a13c406a3a2817ce12b37e8efd6

File: examples/gno.land/r/matijamarjanovic/config/config.gno
Comment (matijamarjanovic) [reply examples/gno.land/r/matijamarjanovic/config/config.gno:27]: Hi, pleas see fix im this commit: 15e15a2429bd67442a8e93f28eeac5ecfef1abeb

Comment (matijamarjanovic) [reply examples/gno.land/r/matijamarjanovic/config/config.gno:27]: Hi, please see fix here: ef98d67f51538ecdde5c496967a83b4d00e94673

File: examples/gno.land/r/matijamarjanovic/config/config.gno
Comment (matijamarjanovic) [reply examples/gno.land/r/matijamarjanovic/config/config.gno:25]: Hi, please see fix here: ef98d67f51538ecdde5c496967a83b4d00e94673

**Review by notJoon — APPROVED** [review]

File: examples/gno.land/r/matijamarjanovic/home/config.gno
Comment (Maelkum) [inline examples/gno.land/r/matijamarjanovic/home/config.gno:1]: Curious, is this a `gno` thing, to have `config` package with this code, and duplicate it in `home`? Can it be a single thing? Are both needed?

File: examples/gno.land/r/matijamarjanovic/home/home.gno
Comment (Maelkum) [inline examples/gno.land/r/matijamarjanovic/home/home.gno:97]: Wouldn't `%v` work instead of `strconv`?

Also you could deduplicate these functions quite a bit.

File: examples/gno.land/r/matijamarjanovic/home/home.gno
Comment (Maelkum) [inline examples/gno.land/r/matijamarjanovic/home/home.gno:130]: Does gno have templates? This could be much cleaner.

Also you have a Sprintf with an empty string there in the second line.

**Review by Maelkum — COMMENTED** [review]
Passerby here, feel free to ignore my comments. Was just passing by and was curious about why some things are the way they are

**Comment by matijamarjanovic** [comment]
Thank you for your review @Maelkum before I say anything I have to say that this is the first thing I have built in Gno and generally on gno chain. This was built by following @leohhhn's live coding session (which is on gno.land's Youtube channel) where he built his own homepage which is in the examples. This is why I chose to do some of the things I did the way I did them. I fixed couple of the things you mentioned that I thought needed fixing the rest I will try explaining in the replies to your comments.

File: examples/gno.land/r/matijamarjanovic/home/config.gno
Comment (matijamarjanovic) [reply examples/gno.land/r/matijamarjanovic/home/config.gno:20]: This is one of the things I did the way Leon did, do not know the reason but I've seen it done this way so I did the same thinig.

File: examples/gno.land/r/matijamarjanovic/home/config.gno
Comment (matijamarjanovic) [reply examples/gno.land/r/matijamarjanovic/home/config.gno:65]: These two functions Leon made so he could show that it can be done in two different ways, although in official documentation the use of panic is encouraged.

File: examples/gno.land/r/matijamarjanovic/home/home.gno
Comment (matijamarjanovic) [reply examples/gno.land/r/matijamarjanovic/home/home.gno]: I made it this way because I thought it's more readable, besides, Leon mentioned that there is a package that will be used to build the Render() function instead of using raw HTML, so in time this will be fixed if needed.

File: examples/gno.land/r/matijamarjanovic/home/config.gno
Comment (matijamarjanovic) [reply examples/gno.land/r/matijamarjanovic/home/config.gno:1]: What I think was the reason for this is showing how to import realms from different packages but also dividing it into 2 logical sections. In the end I put the config realm in the same package as home.gno for easier deployment. Still I think it's better to leave them separate just for simplicity. There is no reason besides this to not put them in one file.

File: examples/gno.land/r/matijamarjanovic/home/home.gno
Comment (matijamarjanovic) [reply examples/gno.land/r/matijamarjanovic/home/home.gno]: Fixed this

File: examples/gno.land/r/matijamarjanovic/home/config.gno
Comment (matijamarjanovic) [reply examples/gno.land/r/matijamarjanovic/home/config.gno:65]: With this I agree, at the time I was uncertain so I decided to lean on the code from the example. Getting fixed in the next commit.

Comment (matijamarjanovic) [reply examples/gno.land/r/matijamarjanovic/home/config.gno:65]: So reading this I almost got confused but I remembered the reason this is the way it is. Like I mentioned [here](https://github.com/gnolang/gno/pull/2916#discussion_r1799594051), there was a discussion on this topic that explains the difference between these functions, there is also a solution to the dilemma included in the discussion, but as this code works as intended for my homepage I decided to not change it and leave it as is - the thing is both my functions used the std.GetPrevRealm().Addr() to make the caller which wouldn't work if the config.gno and home.gno were not in the same package. The way it would work is like it is now - checkAuthorized uses std.GetOrigCaller() and AssertAuthorized uses std.GetPrevRealm().Addr().

File: examples/gno.land/r/matijamarjanovic/home/config.gno
Comment (matijamarjanovic) [reply examples/gno.land/r/matijamarjanovic/home/config.gno]: True, but I don't see a benefit to this besides a code that is a little easier to read.

File: examples/gno.land/r/matijamarjanovic/home/home.gno
Comment (matijamarjanovic) [reply examples/gno.land/r/matijamarjanovic/home/home.gno]: I'm curious, why do you think it makes a significant difference if I don't use `init()`?

File: examples/gno.land/r/matijamarjanovic/home/home.gno
Comment (matijamarjanovic) [reply examples/gno.land/r/matijamarjanovic/home/home.gno]: To my knowledge - (g)no.
Thanks for pointing out, I must have missed the empty string. Fixed it in one of the newer commits.

File: examples/gno.land/r/matijamarjanovic/home/config.gno
Comment (matijamarjanovic) [reply examples/gno.land/r/matijamarjanovic/home/config.gno]: This doesn't make sense because of [this](https://github.com/gnolang/gno/pull/2916#discussion_r1799875477).

File: examples/gno.land/r/matijamarjanovic/home/home.gno
Comment (Maelkum) [reply examples/gno.land/r/matijamarjanovic/home/home.gno]: If you have a variable that you want to initialize, initialize it.
If you have something that you want to DO, and you want to do it before anything else happens, use `init()`.

When reading the code - you jump to a variable and you see its default value (single source of truth).
This is simpler than finding a variable and looking for places where it's initialized, which might be an `init` call that's tucked away somewhere.

In general, don't use `init()` if you don't have a reason to.

File: examples/gno.land/r/matijamarjanovic/home/config.gno
Comment (Maelkum) [reply examples/gno.land/r/matijamarjanovic/home/config.gno:1]: I meant why is there a `home` package and `config` package, where the config/address handling code is duplicated. The config package can be removed, no?

File: examples/gno.land/r/matijamarjanovic/home/config.gno
Comment (leohhhn) [reply examples/gno.land/r/matijamarjanovic/home/config.gno]: Currently, top level initializations & `init` are the same; it can be done either way. For errors, you should use `errors.New` as @Maelkum suggested. 

Also, would be good to explicitly cast the string of the address to `std.Address`

File: examples/gno.land/r/matijamarjanovic/home/config.gno
Comment (matijamarjanovic) [reply examples/gno.land/r/matijamarjanovic/home/config.gno:1]: Yes, I understand now, I believe I fixed this in one of the newer commits.

File: examples/gno.land/r/matijamarjanovic/home/config.gno
Comment (matijamarjanovic) [reply examples/gno.land/r/matijamarjanovic/home/config.gno]: Initialized errors and cast the address in these commits: 3dc2c4e3009ded5948fbaa7c4608fb0e096428d5 and 0c35c090d8f40176785f36f692bab3495255bd75

**Comment by jefft0** [comment]
Removed the "review/triage-pending" label because notJoon approved and this is ready for core dev review.

**Comment by thehowl** [comment]
please run `gno mod tidy` and `make fmt` in `examples` :)

pinging @leohhhn for a review.

File: examples/gno.land/r/matijamarjanovic/home/home_test.gno
Comment (leohhhn) [inline examples/gno.land/r/matijamarjanovic/home/home_test.gno:5]: Formatting completely off. Try using `gno lint` & `gno fmt`

File: examples/gno.land/r/matijamarjanovic/home/home_test.gno
Comment (leohhhn) [inline examples/gno.land/r/matijamarjanovic/home/home_test.gno:11]: formatting

File: examples/gno.land/r/matijamarjanovic/home/home_test.gno
Comment (leohhhn) [inline examples/gno.land/r/matijamarjanovic/home/home_test.gno:16]: Try using uassert & urequire

File: examples/gno.land/r/matijamarjanovic/home/home_test.gno
Comment (leohhhn) [inline examples/gno.land/r/matijamarjanovic/home/home_test.gno:27]: formatting

File: examples/gno.land/r/matijamarjanovic/home/home_test.gno
Comment (leohhhn) [inline examples/gno.land/r/matijamarjanovic/home/home_test.gno:43]: formatting

File: examples/gno.land/r/matijamarjanovic/snakeart/snakeart.gno
Comment (leohhhn) [inline examples/gno.land/r/matijamarjanovic/snakeart/snakeart.gno:1]: I suggest you separate this out in a different PR, as the config & home packages will need to be reworked quite a bit.

**Comment by matijamarjanovic** [comment]
Hey, @leohhhn  and @thehowl thank you for reviewing my code. As HTML is no longer in use on gno.land I will have to change the home realm entirely (voters can no longer change colors). As I implement this I will make sure to update everything you pointed out. I'll try the suggested libraries for .md and provide feedback.

File: examples/gno.land/r/matijamarjanovic/snakeart/snakeart.gno
Comment (matijamarjanovic) [reply examples/gno.land/r/matijamarjanovic/snakeart/snakeart.gno:1]: I don't think it's worthy of saving so I will just leave it out

**Comment by thehowl** [comment]
Putting in draft, click ready for review when it's ready for a second look.

**Comment by Gno2D2** [comment]
I'm a bot that assists the Gno Core team in maintaining this repository. My role is to ensure that contributors understand and follow our guidelines, helping to streamline the development process.

The following requirements must be fulfilled before a pull request can be merged.
Some requirement checks are automated and can be verified by the CI, while others need manual verification by a staff member.

These requirements are defined in this [configuration file](https://github.com/gnolang/gno/tree/master/contribs/github-bot/internal/config/config.go).

## Automated Checks

 🟢 Maintainers must be able to edit this pull request ([more info](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork))
 🟢 The pull request head branch must be up-to-date with its base ([more info](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/keeping-your-pull-request-in-sync-with-the-base-branch))


## Manual Checks

- [ ] The pull request description provides enough details


<details><summary><b>Debug</b></summary><blockquote>
<details><summary><b>Automated Checks</b></summary><blockquote>

<details><summary>Maintainers must be able to edit this pull request (more info)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 The pull request was created from a fork (head branch repo: matijamarjanovic/gno)

```
### Then
```
🟢 Requirement satisfied
└── 🟢 Maintainer can modify this pull request

```
</blockquote></details>

<details><summary>The pull request head branch must be up-to-date with its base (more info)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 On every pull request

```
### Then
```
🟢 Requirement satisfied
└── 🟢 Head branch (matijamarjanovic:master) is up to date with base (master): behind by 0 / ahead by 27

```
</blockquote></details>

</blockquote></details>


<details><summary><b>Manual Checks</b></summary><blockquote>

<details><summary>The pull request description provides enough details</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 Not (🔴 Pull request author is a member of the team: core-contributors)

```
### Can be checked by
 - team core-contributors

</blockquote></details>

</blockquote></details>

</blockquote></details>

File: examples/gno.land/r/matijamarjanovic/home/home.gno
Comment (matijamarjanovic) [reply examples/gno.land/r/matijamarjanovic/home/home.gno:149]: Have reconstructed the homerealm now using markdown, will explain it in more detail in separate comment on this PR, used the first suggested lib and will make sure to leave a comment feedback in their PR

File: examples/gno.land/r/matijamarjanovic/home/home_test.gno
Comment (matijamarjanovic) [reply examples/gno.land/r/matijamarjanovic/home/home_test.gno]: Included in one of the newer commits

File: examples/gno.land/r/matijamarjanovic/home/config.gno
Comment (matijamarjanovic) [reply examples/gno.land/r/matijamarjanovic/home/config.gno]: I always found this topic extremely confusing ever since @stefann-01 wrote his research on it haha. I will fix it in the next commit to GetOrigCaller

**Comment by leohhhn** [comment]
Hey @matijamarjanovic,

Generally you shouldn't take other people's code like this (the MDUI pkg). I know this was not your intention, but what you did was just copy paste this lib and if this PR is merged it will be your commit, effectively denying the original author their rights.

What you should do, is, either use a library that exists in master (`p/moul/md` was merged recently), or make a branch from the branch that contains the code you want to use, which will make your PR dependent on the original PR you branched off of.

I suggest you switch to using `p/moul/md` and I'll take a look. It shouldn't be too big of a change.

**Comment by matijamarjanovic** [comment]
> Hey @matijamarjanovic,
> 
> Generally you shouldn't take other people's code like this (the MDUI pkg). I know this was not your intention, but what you did was just copy paste this lib and if this PR is merged it will be your commit, effectively denying the original author their rights.
> 
> What you should do, is, either use a library that exists in master (`p/moul/md` was merged recently), or make a branch from the branch that contains the code you want to use, which will make your PR dependent on the original PR you branched off of.
> 
> I suggest you switch to using `p/moul/md` and I'll take a look. It shouldn't be too big of a change.

I thught this make be an issue and i agree, I even credited the author in the code through a comment above the code. I'll be commiting the fix in the next hour

**Review by leohhhn — APPROVED** [review]
LGTM 💯

**Comment by leohhhn** [comment]
Just check the CI; for example, you should run `make fmt` in the examples folder.

File: examples/gno.land/r/matijamarjanovic/home/config.gno
Comment (thehowl) [inline examples/gno.land/r/matijamarjanovic/home/config.gno:26]: note: it is idiomatic in gno to panic in these cases, as panicking rolls back the transaction, which is better here

**Review by thehowl — APPROVED** [review]

**Pull request merged**
